src/metrics/system_metrics.rs

```diff
@@ -0,0 +1,96 @@
+use std::sync::{atomic::AtomicU64, Arc, Mutex};
+use prometheus_client::metrics::gauge::Gauge;
+use prometheus_client::registry::Registry;
+use sysinfo::{Disks, Networks, System};
+use tokio::time::{self, Duration};
+
+pub struct SystemMetrics {
+    system: Arc<Mutex<System>>,
+    cpu_usage: Gauge<f64, AtomicU64>,
+    total_memory: Gauge<f64, AtomicU64>,
+    used_memory: Gauge<f64, AtomicU64>,
+    free_memory: Gauge<f64, AtomicU64>,
+    total_disk_available: Gauge<f64, AtomicU64>,
+    total_disk_space: Gauge<f64, AtomicU64>,
+    network_in: Gauge<f64, AtomicU64>,
+    network_out: Gauge<f64, AtomicU64>,
+    load_avg: Gauge<f64, AtomicU64>,
+}
+
+impl SystemMetrics {
+    pub fn new(registry: &mut Registry) -> Self {
+        let cpu_usage = Gauge::<f64, AtomicU64>::default();
+
+        let total_memory = Gauge::<f64, AtomicU64>::default();
+        let used_memory = Gauge::<f64, AtomicU64>::default();
+        let free_memory = Gauge::<f64, AtomicU64>::default();
+
+        let total_disk_space = Gauge::<f64, AtomicU64>::default();
+        let total_disk_available = Gauge::<f64, AtomicU64>::default();
+
+        let network_in = Gauge::<f64, AtomicU64>::default();
+        let network_out = Gauge::<f64, AtomicU64>::default();
+        let load_avg = Gauge::<f64, AtomicU64>::default();
+
+        registry.register("cpu_usage_percent", "CPU usage percent", cpu_usage.clone());
+
+        registry.register("total_memory", "Total memory", total_memory.clone());
+        registry.register("used_memory", "Used memory", used_memory.clone());
+        registry.register("free_memory", "Free memory", free_memory.clone());
+
+        registry.register("total_disk_space", "Total disk space", total_disk_space.clone());
+        registry.register("total_disk_available", "Total disk available", total_disk_available.clone());
+
+        registry.register("network_in_bytes", "Incoming network traffic in bytes", network_in.clone());
+        registry.register("network_out_bytes", "Outgoing network traffic in bytes", network_out.clone());
+
+        registry.register("load_average", "System load average", load_avg.clone());
+
+        let system = Arc::new(Mutex::new(System::new_all()));
+
+        Self { system, cpu_usage, total_memory, used_memory, free_memory, total_disk_available, total_disk_space, network_in, network_out, load_avg }
+    }
+
+    pub async fn start_collecting(self: Arc<Self>) {
+        let mut interval = time::interval(Duration::from_secs(5));
```

Comment:
Can the collection interval be set in the configuration file?

Reply:
Since **sysinfo** only provides separate refresh for CPU, memory, and process metrics; network and disk metrics cannot be refreshed individually.

An approach could be to use **a single interval** to refresh all system metrics, requiring only a single line addition to the .hcl configuration and passing the `config.sys_metrics_collection_interval` to `metrics::SystemMetrics::start_collecting` method.

in src/cli/command/server.rs
```rust
    let metrics_manager = Arc::new(RwLock::new(MetricsManager::new()));
    let system_metrics = Arc::clone(&metrics_manager.read().unwrap().system_metrics);
    let interval = Arc::new(config.sys_metrics_collection_interval);
    ...
    server.block_on(async {
        tokio::spawn(async {
            system_metrics.start_collecting(interval).await;
        });
        http_server.run().await
    })?;
    let _ = server.run();
```

Reply:
> @wa5i
> 
> Decision item: Use a single interval for all system metrics?
> 
> Since current **sysinfo** only provides separate refresh for CPU, memory, and process metrics; network and disk metrics cannot be refreshed individually.

Decision: Interval configuration is supported through the configuration file, but currently limited to a single interval.